http/cors.go

```diff
@@ -0,0 +1,32 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/vault"
+)
+
+func wrapCORSHandler(h http.Handler, core *vault.Core) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		corsConf := core.CORSConfig()
```

Comment:
@jefferai That means that for _every_ non-GET or non-HEAD AJAX request sent to Vault, the browser will send a preflight request (if it does not have one cached) to see if the request it wants to send is going to be accepted by Vault. The browser will *always* require the `Access-Control-Allow-Origin` headers if the Vault server does not exist on the same origin as the request from the browser.